airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_discover.py

```diff
@@ -0,0 +1,226 @@
+from typing import Any, Dict, List, Optional
+from unittest import TestCase, expectedFailure, mock
+from airbyte_cdk import ConfiguredAirbyteCatalog, SyncMode
+from airbyte_cdk.test.entrypoint_wrapper import EntrypointOutput, discover
+from airbyte_cdk.test.catalog_builder import CatalogBuilder
+from airbyte_cdk.test.mock_http import HttpMocker
+from integration.pagination import SnowflakePaginationStrategy
+from integration.response_builder import JsonPath, SnowflakeResponseBuilder, create_response_builder
+from airbyte_protocol.models import (
+    AirbyteStateBlob,
+    AirbyteStateMessage,
+    AirbyteStreamState,
+    ConfiguredAirbyteCatalog,
+    FailureType,
+    StreamDescriptor,
+    SyncMode,
+)
+from airbyte_cdk.test.mock_http.response_builder import (
+    FieldPath,
+    Path,
+    RecordBuilder,
+    create_record_builder,
+    find_template,
+)
+from parameterized import parameterized
+
+from integration.config import ConfigBuilder
+from integration.request_builder import SnowflakeRequestBuilder
+from integration.response_builder import JsonPath
+from source_snowflake.source import SourceSnowflake
+
+
+
+_WAREHOUSE = "WAREHOUSE"
+_DATABASE = "DATABASE"
+_SCHEMA = "INTEGRATION_TEST"
+_TABLE = "TEST_TABLE"
+_ROLE = "ROLE"
+_HOST = "host.com"
+_REQUESTID = "123"
+_HANDLE = "123-DUMMY-HANDLE-123"
+
+
+def _config() -> ConfigBuilder:
+    return ConfigBuilder(jwt_token="123", host=_HOST, schema=_SCHEMA, database=_DATABASE, role=_ROLE, warehouse=_WAREHOUSE)
+
+
+def table_request() -> SnowflakeRequestBuilder:
+    return SnowflakeRequestBuilder.statement_endpoint(host=_HOST, schema=_SCHEMA, database=_DATABASE, role=_ROLE, warehouse=_WAREHOUSE)
+
+
+def _catalog(sync_mode: SyncMode) -> ConfiguredAirbyteCatalog:
+    return CatalogBuilder().build()
+
+
+def _source(catalog: ConfiguredAirbyteCatalog, config: Dict[str, Any], state: Optional[List[AirbyteStateMessage]]) -> SourceSnowflake:
```

Comment:
Should we remove those parameters as they are not used? 

Should the instantiation logic be common to the different tests? I see the same code with the same error in `test_table`

Reply:
you're right, as ismail is working on those function. i didn't want to break things